pkg/client/client.go

```diff
@@ -106,6 +115,42 @@ func New(apiToken string, opts ...ClientOption) (*Client, error) {
 	return swoClient, nil
 }
 
+func (c *Client) Do(req *http.Request) (*http.Response, error) {
+	operation := func() (*http.Response, error) {
+		var body []byte
+		var err error
+		if req.Body != nil {
+			body, err = io.ReadAll(req.Body)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// Reset the request body for retries
+		req.Body = io.NopCloser(bytes.NewReader(body))
```

Comment:
Okay. Now for the big one. :)

You're not retrying the request. You're just retrying to read the body which is going to produce the same result every time. You need to be retrying the actual request.